src/utils/file.cpp

```diff
@@ -544,7 +544,9 @@ void OutputFile::EnableFlushing() {
   TryFlushing();
 }
 
-std::pair<const uint8_t *, size_t> OutputFile::CurrentBuffer() const { return {buffer_, buffer_position_.load()}; }
+std::pair<const uint8_t *, size_t> OutputFile::DurableBuffer() const {
+  return {buffer_, buffer_position_.load() - written_since_last_sync_};
```

Comment:
Not safe, `buffer_position_` and `written_since_last_sync_` are not atomically written together/read together. Just maintain a new position which is updated only on sync.